Replace `klog` with `zerolog` for human-friendly logging in Lighthouse Agent

davidohana opened the pull request:
![image](https://user-images.githubusercontent.com/27995472/155492526-f43052e3-165d-4c81-af72-49906db00033.png)

Using `kzerolog` package from `admiral`
All log prints in the agent  were translated manually to use the  [logr](https://github.com/go-logr/logr) interface that is used in [contoller-runtime](https://github.com/kubernetes-sigs/controller-runtime/blob/master/TMP-LOGGING.md).
This required some adaptations since `logr` API is minimalistic and structured (no FATAL level, no WARN level, no format strings..)

Note: some internal k8s packages uses `klog` directly, so few logs may still be printed in the klog format as seen in the image above.

skitt reviewed (changes requested):
The changes themselves seem fine to me, thanks!

However, before this can be merged, please squash your changes into a single commit, and include your explanation from the PR in the commit message itself. As per the guidelines, the commit message should start with “Replace” rater than “Replaced”.

davidohana commented:
> However, before this can be merged, please squash your changes into a single commit, and include your explanation from the PR in the commit message itself. As per the guidelines, the commit message should start with “Replace” rater than “Replaced”.

Thanks. I will squash and comment as requested, once all reviewers finish.
BTW, isn't it simpler to "squash and merge" in GitHub UI when you accept the PR?

skitt commented:
> Thanks. I will squash and comment as requested, once all reviewers finish. BTW, isn't it simpler to "squash and merge" in GitHub UI when you accept the PR?

That relies on whoever accepts the PR remembering to do so, and it regularly happens that people forget!

davidohana committed:
Replace klog with zerolog for human-friendly logging in Lighthouse Agent

Using kzerolog package from Admiral. All log prints in the agent were
translated manually to use the logr interface that is used in
contoller-runtime. This required some adaptations since logr API is
minimalistic and structured (no FATAL level, no WARN level,
no format strings..)

Note: some internal k8s packages uses klog directly, so few logs may
still be printed in the klog format.

Signed-off-by: David Ohana <david.ohana@ibm.com>

skitt reviewed (changes requested):
Given that these changes are widespread, I would rather wait until we start work on 0.13. The risk is that this could end up on the critical path for backports into 0.12 now.

skitt commented:
@davidohana can you take care of rebasing this and fixing the conflicts (without adding a new commit), or would you prefer one of us to handle it?

tpantelis commented:
> @davidohana can you take care of rebasing this and fixing the conflicts (without adding a new commit), or would you prefer one of us to handle it?

I'm in the process of doing it.

tpantelis commented:
> > @davidohana can you take care of rebasing this and fixing the conflicts (without adding a new commit), or would you prefer one of us to handle it?
> 
> I'm in the process of doing it.

I don't have permission to push:

`ERROR: Permission to axon-net/lighthouse.git denied to tpantelis.`

I guess I'd have to push to a new branch/PR.

dfarrell07 commented:
> ERROR: Permission to axon-net/lighthouse.git denied to tpantelis.

I guess it's because the PR is from a branch on a different org vs a branch of the user's fork who's sending the PR.

dfarrell07 commented:
@davidohana Can you rebase this? Or set the configuration to allow Owners to push to this PR's branch?

davidohana commented:
> @davidohana Can you rebase this? Or set the configuration to allow Owners to push to this PR's branch?

I will ask @donaldh to give @tpantelis and @dfarrell07 write permissions to this repo

davidohana commented:
@tpantelis and @dfarrell07 you should now have write permissions to the repo

tpantelis commented:
> @tpantelis and @dfarrell07 you should now have write permissions to the repo

I still get a permission error trying to push to axon-submariner. I pushed the commit to a new PR https://github.com/submariner-io/lighthouse/pull/728.
